core/server/models/post.js

```diff
@@ -164,18 +165,22 @@ Post = ghostBookshelf.Model.extend({
             }
         }
 
-        // keep tags for 'saved' event and deduplicate upper/lowercase tags
-        this.myTags = [];
-
-        _.each(tagsToCheck, function each(item) {
-            for (i = 0; i < self.myTags.length; i = i + 1) {
-                if (self.myTags[i].name.toLocaleLowerCase() === item.name.toLocaleLowerCase()) {
-                    return;
+        // If we have a tags property passed in
+        if (!_.isUndefined(tagsToCheck) && !_.isNull(tagsToCheck)) {
```

Comment:
Added `&& !_.isNull()` here, to also catch the null case at the model layer. This makes it consistent with the API layer & follows on from the conversation in #dev about only allowing empty arrays to delete all tags being much safer.